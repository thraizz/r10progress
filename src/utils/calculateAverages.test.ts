import testdata from "../testdata.json";

import { expect, test } from "vitest";
import { calculateAverages } from "./calculateAverages";

test("should return an object with a Driver key", () => {
  // @ts-expect-error - testdata is a valid Sessions object
  const result = calculateAverages(testdata);
  expect(result).toMatchSnapshot();
});
