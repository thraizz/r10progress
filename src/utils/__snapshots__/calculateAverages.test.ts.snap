// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`calculateAverages > should return averages 1`] = `
[
  {
    "Abflugrichtung": 7.817097081857569,
    "Abflugwinkel": 39.474718654856964,
    "Anstellwinkel": -1.1867991405374863,
    "Backspin": 4413.6926520852485,
    "Ballgeschwindigkeit": 86.1693320840345,
    "Carry-Abweichungsdistanz": 5.017193398054908,
    "Carry-Abweichungswinkel": 8.458575799184686,
    "Carry-Distanz": 44.758916181676526,
    "Drehachse": -0.8371737722088309,
    "Drehrate": 4580.740033318015,
    "Gesamtabweichungsdistanz": 5.7081323441337135,
    "Gesamtabweichungswinkel": 8.387062966823578,
    "Gesamtstrecke": 49.30401970358456,
    "Höhe des Scheitelpunkts": 12.941313294803395,
    "Luftdichte": 1.2268371999999999,
    "Luftdruck": 99.65117000000002,
    "Relative Luftfeuchtigkeit": 76,
    "Schl.gsch.": 101.42322989664054,
    "Schlagfläche": 5.335503735963036,
    "Schlagflächenstellung": 13.421189332709593,
    "Schwungbahn": -3.3884882856817806,
    "Sidespin": -417.0895747016458,
    "Smash Factor": 0.8556477473496203,
    "Temperatur": 8.888889000000002,
    "count": 17,
    "name": "Sand Wedge",
  },
  {
    "Abflugrichtung": 0.5827732823256936,
    "Abflugwinkel": 23.051712020167283,
    "Anstellwinkel": 1.1506810953308428,
    "Backspin": 4352.680712018694,
    "Ballgeschwindigkeit": 117.54561129214537,
    "Carry-Abweichungsdistanz": -2.382613948413304,
    "Carry-Abweichungswinkel": -0.3423297799059323,
    "Carry-Distanz": 66.89559582301548,
    "Drehachse": -3.1393814448799406,
    "Drehrate": 4473.342908586775,
    "Gesamtabweichungsdistanz": -2.259184862886156,
    "Gesamtabweichungswinkel": -0.38063143619469236,
    "Gesamtstrecke": 74.35492747170585,
    "Höhe des Scheitelpunkts": 12.909999420267663,
    "Luftdichte": 1.234700235714285,
    "Luftdruck": 100.06870214285719,
    "Relative Luftfeuchtigkeit": 77.78571428571429,
    "Schl.gsch.": 111.87020209822792,
    "Schlagfläche": -1.9981577428323882,
    "Schlagflächenstellung": 3.7865974477359226,
    "Schwungbahn": -2.670355164978121,
    "Sidespin": 162.90602554593767,
    "Smash Factor": 1.0496038724421992,
    "Temperatur": 8.293650892857146,
    "count": 28,
    "name": "Pitching Wedge",
  },
  {
    "Abflugrichtung": -0.4696579939789242,
    "Abflugwinkel": 13.425719915992683,
    "Anstellwinkel": -0.9158614263869822,
    "Backspin": 4215.018914116754,
    "Ballgeschwindigkeit": 137.35091646642366,
    "Carry-Abweichungsdistanz": -5.335378007756339,
    "Carry-Abweichungswinkel": -1.7396177897850673,
    "Carry-Distanz": 81.34114747577243,
    "Drehachse": -3.658443859157463,
    "Drehrate": 4386.345079210069,
    "Gesamtabweichungsdistanz": -5.8666300773620605,
    "Gesamtabweichungswinkel": -1.8680043121178944,
    "Gesamtstrecke": 97.17075951894124,
    "Höhe des Scheitelpunkts": 8.909910046392017,
    "Luftdichte": 1.2366223111111103,
    "Luftdruck": 100.17076555555559,
    "Relative Luftfeuchtigkeit": 78.22222222222223,
    "Schl.gsch.": 116.7823768196535,
    "Schlagfläche": -1.521172803412709,
    "Schlagflächenstellung": 0.6702350808514489,
    "Schwungbahn": -1.0895022004842758,
    "Sidespin": 223.2393328216341,
    "Smash Factor": 1.1747173218368996,
    "Temperatur": 8.148148244444448,
    "count": 36,
    "name": "6 Iron",
  },
  {
    "Abflugrichtung": 1.926966573132409,
    "Abflugwinkel": 15.135783533917532,
    "Anstellwinkel": -1.1857004306382604,
    "Backspin": 4621.155019802517,
    "Ballgeschwindigkeit": 134.76729867597945,
    "Carry-Abweichungsdistanz": -0.16316428648100958,
    "Carry-Abweichungswinkel": 1.0715296745300293,
    "Carry-Distanz": 82.54763145446778,
    "Drehachse": -2.0749586704704495,
    "Drehrate": 4693.218630642361,
    "Gesamtabweichungsdistanz": 0.2135619772805108,
    "Gesamtabweichungswinkel": 0.9892588668399387,
    "Gesamtstrecke": 97.3018485599094,
    "Höhe des Scheitelpunkts": 10.057312718696064,
    "Luftdichte": 1.2390065688888887,
    "Luftdruck": 100.30643711111111,
    "Relative Luftfeuchtigkeit": 81.57777777777778,
    "Schl.gsch.": 117.50878274523454,
    "Schlagfläche": 1.2794463779156406,
    "Schlagflächenstellung": 1.850954035255644,
    "Schwungbahn": 0.18552126677499878,
    "Sidespin": 227.9454793718126,
    "Smash Factor": 1.146739124703454,
    "Temperatur": 7.962963055555558,
    "count": 45,
    "name": "7 Iron",
  },
  {
    "Abflugrichtung": 2.9818058158900285,
    "Abflugwinkel": 21.631891070185482,
    "Anstellwinkel": -0.16851832173966072,
    "Backspin": 4798.746186127534,
    "Ballgeschwindigkeit": 131.63552798472034,
    "Carry-Abweichungsdistanz": 0.43372803120999726,
    "Carry-Abweichungswinkel": 2.0641729316195927,
    "Carry-Distanz": 88.00635992514121,
    "Drehachse": -2.3247106163888365,
    "Drehrate": 4878.826384673247,
    "Gesamtabweichungsdistanz": 0.5689360579928836,
    "Gesamtabweichungswinkel": 1.989696699219781,
    "Gesamtstrecke": 97.96674439713762,
    "Höhe des Scheitelpunkts": 14.920767286339322,
    "Luftdichte": 1.2386661729729727,
    "Luftdruck": 100.28979702702705,
    "Relative Luftfeuchtigkeit": 81.94594594594595,
    "Schl.gsch.": 119.01466877085106,
    "Schlagfläche": 0.5083725282771362,
    "Schlagflächenstellung": 4.952448735365996,
    "Schwungbahn": -1.1915788349468965,
    "Sidespin": 203.01069940103068,
    "Smash Factor": 1.1083594151817955,
    "Temperatur": 7.987988081081083,
    "count": 37,
    "name": "8 Iron",
  },
  {
    "Abflugrichtung": 2.146061303048599,
    "Abflugwinkel": 21.10096080564871,
    "Anstellwinkel": 1.4025974436095212,
    "Backspin": 4720.932810713605,
    "Ballgeschwindigkeit": 124.36907261074907,
    "Carry-Abweichungsdistanz": 0.5916512845674666,
    "Carry-Abweichungswinkel": 1.7872902343912822,
    "Carry-Distanz": 77.86733227241331,
    "Drehachse": -0.6941298972360972,
    "Drehrate": 4782.152013266958,
    "Gesamtabweichungsdistanz": 1.0908769957902955,
    "Gesamtabweichungswinkel": 1.7728475201420668,
    "Gesamtstrecke": 89.08083576109351,
    "Höhe des Scheitelpunkts": 12.791979577256049,
    "Luftdichte": 1.2391392975609754,
    "Luftdruck": 100.31152658536587,
    "Relative Luftfeuchtigkeit": 81,
    "Schl.gsch.": 113.63369695579347,
    "Schlagfläche": 0.821171174325594,
    "Schlagflächenstellung": 3.9852459605147197,
    "Schwungbahn": -1.191179007927819,
    "Sidespin": 62.92308607334044,
    "Smash Factor": 1.096536422585329,
    "Temperatur": 7.953929631707319,
    "count": 41,
    "name": "9 Iron",
  },
  {
    "Abflugrichtung": 2.6810461937316825,
    "Abflugwinkel": 6.713870685015406,
    "Anstellwinkel": -2.8304858758513416,
    "Backspin": 3011.1894345964706,
    "Ballgeschwindigkeit": 156.00570062440565,
    "Carry-Abweichungsdistanz": 0.973887207784823,
    "Carry-Abweichungswinkel": 2.073704286877598,
    "Carry-Distanz": 78.47512790134975,
    "Drehachse": -0.7349244284310511,
    "Drehrate": 3124.3828321184433,
    "Gesamtabweichungsdistanz": 1.2424707705421107,
    "Gesamtabweichungswinkel": 1.9787290798766273,
    "Gesamtstrecke": 109.81193062237331,
    "Höhe des Scheitelpunkts": 4.223984266764352,
    "Luftdichte": 1.2422471285714283,
    "Luftdruck": 100.36729796428575,
    "Relative Luftfeuchtigkeit": 83.03571428571429,
    "Schl.gsch.": 126.38950820980912,
    "Schlagfläche": 3.1324202407683646,
    "Schlagflächenstellung": 2.2692569621971677,
    "Schwungbahn": 0.8631633184850216,
    "Sidespin": -13.658791490963527,
    "Smash Factor": 1.2341079649012905,
    "Temperatur": 7.410714310714286,
    "count": 56,
    "name": "2 Hybrid",
  },
  {
    "Abflugrichtung": 8.224618752797445,
    "Abflugwinkel": 6.679250664181179,
    "Anstellwinkel": -6.2536665598551435,
    "Backspin": 3591.503465440538,
    "Ballgeschwindigkeit": 146.97816993453307,
    "Carry-Abweichungsdistanz": 10.591983026928371,
    "Carry-Abweichungswinkel": 9.257365730073717,
    "Carry-Distanz": 67.45197168986003,
    "Drehachse": 3.4349730014801025,
    "Drehrate": 3689.654201931424,
    "Gesamtabweichungsdistanz": 15.305640326605904,
    "Gesamtabweichungswinkel": 9.509563048680624,
    "Gesamtstrecke": 103.26430002848308,
    "Höhe des Scheitelpunkts": 3.406023939450582,
    "Luftdichte": 1.2293817,
    "Luftdruck": 99.48521400000001,
    "Relative Luftfeuchtigkeit": 87,
    "Schl.gsch.": 122.1909330423425,
    "Schlagfläche": 9.81582776705424,
    "Schlagflächenstellung": 8.523741907543606,
    "Schwungbahn": 1.2920858272247844,
    "Sidespin": -307.9520378112793,
    "Smash Factor": 1.2028209602168496,
    "Temperatur": 7.7777777,
    "count": 9,
    "name": "4 Wood",
  },
  {
    "Abflugrichtung": 5.858426945824777,
    "Abflugwinkel": 10.55908497687309,
    "Anstellwinkel": -4.774006086011087,
    "Backspin": 5732.660189720892,
    "Ballgeschwindigkeit": 158.84794693037045,
    "Carry-Abweichungsdistanz": 4.994534850120544,
    "Carry-Abweichungswinkel": 4.783994304557001,
    "Carry-Distanz": 105.70661163330078,
    "Drehachse": -3.744422167058914,
    "Drehrate": 5848.421697801159,
    "Gesamtabweichungsdistanz": 5.520343457498858,
    "Gesamtabweichungswinkel": 4.646164023107098,
    "Gesamtstrecke": 121.0299813055223,
    "Höhe des Scheitelpunkts": 10.171980111829695,
    "Luftdichte": 1.2293817,
    "Luftdruck": 99.48521399999996,
    "Relative Luftfeuchtigkeit": 87,
    "Schl.gsch.": 127.9939721932862,
    "Schlagfläche": 5.738359485903094,
    "Schlagflächenstellung": 5.984301569000367,
    "Schwungbahn": 0.9960235578398551,
    "Sidespin": 112.2739451931369,
    "Smash Factor": 1.2411573924844308,
    "Temperatur": 7.777777700000001,
    "count": 31,
    "name": "7 Wood",
  },
  {
    "Abflugrichtung": 3.1852817632581876,
    "Abflugwinkel": 12.70032414664393,
    "Anstellwinkel": -0.0976412859947785,
    "Backspin": 3236.360899552055,
    "Ballgeschwindigkeit": 179.23176222638733,
    "Carry-Abweichungsdistanz": -0.44162951021090796,
    "Carry-Abweichungswinkel": 0.297686825627866,
    "Carry-Distanz": 129.58500323088273,
    "Drehachse": -12.93587657160942,
    "Drehrate": 3363.6729019828463,
    "Gesamtabweichungsdistanz": -2.9090115136426427,
    "Gesamtabweichungswinkel": -0.27518958565981494,
    "Gesamtstrecke": 152.1266622128694,
    "Höhe des Scheitelpunkts": 13.533535511597343,
    "Luftdichte": 1.2352991260869557,
    "Luftdruck": 99.89082104347821,
    "Relative Luftfeuchtigkeit": 85.17391304347827,
    "Schl.gsch.": 139.85683103839304,
    "Schlagfläche": 3.471998462534469,
    "Schlagflächenstellung": 1.5271788576374883,
    "Schwungbahn": 1.9448195697193074,
    "Sidespin": 476.44047248737394,
    "Smash Factor": 1.2830033243295882,
    "Temperatur": 7.608695621739131,
    "count": 46,
    "name": "Driver",
  },
]
`;

exports[`calculateAverages > should return less strokes when using IQR 1`] = `
[
  {
    "Abflugrichtung": 5.495011917182377,
    "Abflugwinkel": 39.474718654856964,
    "Anstellwinkel": -1.8035897985100746,
    "Backspin": 4413.6926520852485,
    "Ballgeschwindigkeit": 86.1693320840345,
    "Carry-Abweichungsdistanz": 6.140889998525381,
    "Carry-Abweichungswinkel": 5.890321991273335,
    "Carry-Distanz": 44.758916181676526,
    "Drehachse": -0.8371737722088309,
    "Drehrate": 4580.740033318015,
    "Gesamtabweichungsdistanz": 5.575742840766907,
    "Gesamtabweichungswinkel": 5.731286248990467,
    "Gesamtstrecke": 49.30401970358456,
    "Höhe des Scheitelpunkts": 12.941313294803395,
    "Luftdichte": 1.2268371999999999,
    "Luftdruck": 99.65117000000002,
    "Relative Luftfeuchtigkeit": 76,
    "Schl.gsch.": 101.42322989664054,
    "Schlagfläche": 5.465428785483042,
    "Schlagflächenstellung": 11.410844888538122,
    "Schwungbahn": -3.3884882856817806,
    "Sidespin": -417.0895747016458,
    "Smash Factor": 0.8556477473496203,
    "Temperatur": 8.888889000000002,
    "count": 17,
    "name": "Sand Wedge",
  },
  {
    "Abflugrichtung": -2.749442399442196,
    "Abflugwinkel": 23.051712020167283,
    "Anstellwinkel": 1.1577677824534476,
    "Backspin": 4352.680712018694,
    "Ballgeschwindigkeit": 117.54561129214537,
    "Carry-Abweichungsdistanz": -2.439530277252197,
    "Carry-Abweichungswinkel": -4.368345571756363,
    "Carry-Distanz": 66.89559582301548,
    "Drehachse": -3.662611685693264,
    "Drehrate": 4473.342908586775,
    "Gesamtabweichungsdistanz": -6.277689542770386,
    "Gesamtabweichungswinkel": -4.503243777751923,
    "Gesamtstrecke": 74.35492747170585,
    "Höhe des Scheitelpunkts": 12.909999420267663,
    "Luftdichte": 1.2347002357142858,
    "Luftdruck": 100.06870214285716,
    "Relative Luftfeuchtigkeit": 77.78571428571429,
    "Schl.gsch.": 111.87020209822792,
    "Schlagfläche": -3.9276850163936614,
    "Schlagflächenstellung": -0.04733739449427678,
    "Schwungbahn": -2.670355164978121,
    "Sidespin": 260.3111279099076,
    "Smash Factor": 1.0401032268469852,
    "Temperatur": 8.293650892857146,
    "count": 28,
    "name": "Pitching Wedge",
  },
  {
    "Abflugrichtung": -2.0066645927727222,
    "Abflugwinkel": 14.200762264868793,
    "Anstellwinkel": -0.9158614263869822,
    "Backspin": 4041.2453685087316,
    "Ballgeschwindigkeit": 138.586160421738,
    "Carry-Abweichungsdistanz": -5.335378007756339,
    "Carry-Abweichungswinkel": -2.7928868072373527,
    "Carry-Distanz": 85.52456300398883,
    "Drehachse": -3.658443859157463,
    "Drehrate": 4207.105669806985,
    "Gesamtabweichungsdistanz": -5.8666300773620605,
    "Gesamtabweichungswinkel": -2.9411656413759504,
    "Gesamtstrecke": 103.84805783358487,
    "Höhe des Scheitelpunkts": 8.909910046392017,
    "Luftdichte": 1.2366223111111105,
    "Luftdruck": 100.17076555555556,
    "Relative Luftfeuchtigkeit": 78.22222222222223,
    "Schl.gsch.": 116.7823768196535,
    "Schlagfläche": -3.1767011016847624,
    "Schlagflächenstellung": -1.360888984054327,
    "Schwungbahn": -0.9381471614042918,
    "Sidespin": 223.2393328216341,
    "Smash Factor": 1.1834806898041161,
    "Temperatur": 8.148148244444448,
    "count": 36,
    "name": "6 Iron",
  },
  {
    "Abflugrichtung": 1.3539007915691896,
    "Abflugwinkel": 15.46841497854753,
    "Anstellwinkel": -1.6545524522662163,
    "Backspin": 4250.326318650019,
    "Ballgeschwindigkeit": 134.76729867597945,
    "Carry-Abweichungsdistanz": -0.16316428648100958,
    "Carry-Abweichungswinkel": 0.4048289277336814,
    "Carry-Distanz": 85.5548688755479,
    "Drehachse": -2.9374557183611962,
    "Drehrate": 4324.171375093006,
    "Gesamtabweichungsdistanz": 0.2135619772805108,
    "Gesamtabweichungswinkel": 0.3002523888241161,
    "Gesamtstrecke": 99.89404767058616,
    "Höhe des Scheitelpunkts": 10.057312718696064,
    "Luftdichte": 1.239006568888889,
    "Luftdruck": 100.30643711111108,
    "Relative Luftfeuchtigkeit": 81.57777777777778,
    "Schl.gsch.": 117.50878274523454,
    "Schlagfläche": 1.2794463779156406,
    "Schlagflächenstellung": 0.9660381193865429,
    "Schwungbahn": 0.1991019051794599,
    "Sidespin": 227.9454793718126,
    "Smash Factor": 1.1467391247034537,
    "Temperatur": 7.962963055555558,
    "count": 45,
    "name": "7 Iron",
  },
  {
    "Abflugrichtung": 0.42130612099871917,
    "Abflugwinkel": 22.1237168179618,
    "Anstellwinkel": -0.16851832173966072,
    "Backspin": 4649.753446691177,
    "Ballgeschwindigkeit": 134.2713560998625,
    "Carry-Abweichungsdistanz": 0.43372803120999726,
    "Carry-Abweichungswinkel": -0.9103447058621574,
    "Carry-Distanz": 90.9748035975865,
    "Drehachse": -3.7277559249690086,
    "Drehrate": 4736.416022805606,
    "Gesamtabweichungsdistanz": 0.5689360579928836,
    "Gesamtabweichungswinkel": -1.010434329509735,
    "Gesamtstrecke": 99.98033502366808,
    "Höhe des Scheitelpunkts": 14.920767286339322,
    "Luftdichte": 1.2386661729729733,
    "Luftdruck": 100.28979702702698,
    "Relative Luftfeuchtigkeit": 81.94594594594595,
    "Schl.gsch.": 119.78717471368637,
    "Schlagfläche": 0.5083725282771362,
    "Schlagflächenstellung": 1.180139506564421,
    "Schwungbahn": -1.1915788349468965,
    "Sidespin": 199.6110701424735,
    "Smash Factor": 1.1258248615225928,
    "Temperatur": 7.987988081081083,
    "count": 37,
    "name": "8 Iron",
  },
  {
    "Abflugrichtung": 0.010665227315927806,
    "Abflugwinkel": 21.60557093024254,
    "Anstellwinkel": 1.4025974436095212,
    "Backspin": 4811.389404296875,
    "Ballgeschwindigkeit": 125.66232976340773,
    "Carry-Abweichungsdistanz": -1.5041306121036029,
    "Carry-Abweichungswinkel": -0.23567758615200335,
    "Carry-Distanz": 81.73265918932464,
    "Drehachse": -2.268187840684102,
    "Drehrate": 4874.077203369141,
    "Gesamtabweichungsdistanz": -1.231434824375006,
    "Gesamtabweichungswinkel": -0.28489440985215014,
    "Gesamtstrecke": 90.37949151992798,
    "Höhe des Scheitelpunkts": 12.791979577256049,
    "Luftdichte": 1.2391392975609754,
    "Luftdruck": 100.31152658536583,
    "Relative Luftfeuchtigkeit": 81,
    "Schl.gsch.": 113.63369695579347,
    "Schlagfläche": -0.8410910105234698,
    "Schlagflächenstellung": 0.8996537359137284,
    "Schwungbahn": -1.7935291313781188,
    "Sidespin": 187.88381023896045,
    "Smash Factor": 1.1035815823851647,
    "Temperatur": 7.953929631707319,
    "count": 41,
    "name": "9 Iron",
  },
  {
    "Abflugrichtung": 1.9045827802684572,
    "Abflugwinkel": 6.296413965658708,
    "Anstellwinkel": -2.8304858758513416,
    "Backspin": 2956.6006913618608,
    "Ballgeschwindigkeit": 157.0188255051721,
    "Carry-Abweichungsdistanz": 0.973887207784823,
    "Carry-Abweichungswinkel": 1.5950032077052376,
    "Carry-Distanz": 78.47512790134975,
    "Drehachse": 0.42168186842040584,
    "Drehrate": 3029.4182332356772,
    "Gesamtabweichungsdistanz": 1.2424707705421107,
    "Gesamtabweichungswinkel": 1.4632529540495438,
    "Gesamtstrecke": 120.06156158447266,
    "Höhe des Scheitelpunkts": 4.024687428636985,
    "Luftdichte": 1.2422471285714278,
    "Luftdruck": 100.36729796428583,
    "Relative Luftfeuchtigkeit": 83.03571428571429,
    "Schl.gsch.": 126.38950820980912,
    "Schlagfläche": 2.2017886627603462,
    "Schlagflächenstellung": 0.9406376199902229,
    "Schwungbahn": 0.8631633184850216,
    "Sidespin": 89.87537359723858,
    "Smash Factor": 1.2415565832962405,
    "Temperatur": 7.410714310714286,
    "count": 56,
    "name": "2 Hybrid",
  },
  {
    "Abflugrichtung": 8.224618752797445,
    "Abflugwinkel": 6.679250664181179,
    "Anstellwinkel": -6.2536665598551435,
    "Backspin": 3591.503465440538,
    "Ballgeschwindigkeit": 146.97816993453307,
    "Carry-Abweichungsdistanz": 10.591983026928371,
    "Carry-Abweichungswinkel": 9.257365730073717,
    "Carry-Distanz": 71.84153079986572,
    "Drehachse": 3.4349730014801025,
    "Drehrate": 3689.654201931424,
    "Gesamtabweichungsdistanz": 15.305640326605904,
    "Gesamtabweichungswinkel": 9.509563048680624,
    "Gesamtstrecke": 103.26430002848308,
    "Höhe des Scheitelpunkts": 2.7453381568193436,
    "Luftdichte": 1.2293817,
    "Luftdruck": 99.48521400000001,
    "Relative Luftfeuchtigkeit": 87,
    "Schl.gsch.": 123.7121316562193,
    "Schlagfläche": 9.81582776705424,
    "Schlagflächenstellung": 8.523741907543606,
    "Schwungbahn": 1.2920858272247844,
    "Sidespin": -307.9520378112793,
    "Smash Factor": 1.1768573700582714,
    "Temperatur": 7.7777777,
    "count": 9,
    "name": "4 Wood",
  },
  {
    "Abflugrichtung": 4.996504805485407,
    "Abflugwinkel": 10.8193590482076,
    "Anstellwinkel": -5.279462419707199,
    "Backspin": 5732.660189720892,
    "Ballgeschwindigkeit": 158.84794693037045,
    "Carry-Abweichungsdistanz": 4.994534850120544,
    "Carry-Abweichungswinkel": 3.7042376826206844,
    "Carry-Distanz": 105.70661163330078,
    "Drehachse": -5.985707176656558,
    "Drehrate": 5848.421697801159,
    "Gesamtabweichungsdistanz": 5.520343457498858,
    "Gesamtabweichungswinkel": 3.4999159077803292,
    "Gesamtstrecke": 121.0299813055223,
    "Höhe des Scheitelpunkts": 10.171980111829695,
    "Luftdichte": 1.2293817,
    "Luftdruck": 99.48521399999996,
    "Relative Luftfeuchtigkeit": 87,
    "Schl.gsch.": 128.8152333276739,
    "Schlagfläche": 5.738359485903094,
    "Schlagflächenstellung": 4.9245359122753145,
    "Schwungbahn": 0.5676449362573952,
    "Sidespin": 202.3300086657206,
    "Smash Factor": 1.2411573924844304,
    "Temperatur": 7.777777700000001,
    "count": 31,
    "name": "7 Wood",
  },
  {
    "Abflugrichtung": 2.802935419480006,
    "Abflugwinkel": 11.823284820068714,
    "Anstellwinkel": -0.0976412859947785,
    "Backspin": 2739.810972667876,
    "Ballgeschwindigkeit": 180.5810309828378,
    "Carry-Abweichungsdistanz": -0.44162951021090796,
    "Carry-Abweichungswinkel": 0.297686825627866,
    "Carry-Distanz": 129.58500323088273,
    "Drehachse": -12.93587657160942,
    "Drehrate": 2801.1510188405105,
    "Gesamtabweichungsdistanz": -2.9090115136426427,
    "Gesamtabweichungswinkel": -0.27518958565981494,
    "Gesamtstrecke": 153.94173346625433,
    "Höhe des Scheitelpunkts": 13.533535511597343,
    "Luftdichte": 1.2352991260869566,
    "Luftdruck": 99.89082104347827,
    "Relative Luftfeuchtigkeit": 85.17391304347827,
    "Schl.gsch.": 139.85683103839304,
    "Schlagfläche": 2.5022257014431735,
    "Schlagflächenstellung": 0.2917546033859253,
    "Schwungbahn": 2.231783861718658,
    "Sidespin": 476.44047248737394,
    "Smash Factor": 1.311382414768214,
    "Temperatur": 7.608695621739131,
    "count": 46,
    "name": "Driver",
  },
]
`;
